### ISSUE TITLE
When on Windows, do as Windows does.

### ISSUE THREAD

**Issue by MicahZoltu** [moby/moby#40848]
https://github.com/moby/moby/blob/f6a5ccf492e8eab969ffad8404117806b4a15a35/pkg/homedir/homedir_windows.go#L10

**TL;DR**: Use correct folders for writing configuration and cache files.  If you are targeting Go 1.12+ then just use `os.UserConfigDir` or `os.UserCacheDir` as appropriate.  If lower than Go 1.12, then you can copy the logic from Go source code: https://github.com/golang/go/blob/79395c55e27618c1d48218274498d8390705979c/src/os/file.go#L362-L462

----

In Windows, the root of the user directory (`%USERPROFILE%`) is NOT the right place to store application settings. This is a very common convention on Linux, but Windows != Linux and not all Windows users want to pretend they are running Linux. On Windows, the correct location for application settings is `%APPDATA%` and the correct location for extensions is `%LOCALAPPDATA%` or `%PROGRAMDATA%` (depending on level of security consciousness).

`%USERPROFILE` is the root of the user's space on the hard drive, and it is meant to hold documents that the user explicitly writes there (e.g., via a save dialog box) and to hold top-level folders for organization such as `Documents`, `Videos`, `Pictures`, and the appropriately hidden `AppData` folder (which is the root of the tree for applications storing their stuff).

Along with the organization problems that is caused by storing files in `%USERPROFILE%`, it also presents problems with roaming profiles and backup scripts as Windows and Windows tools do not automatically sync/backup everything in `%USERPROFILE%`.  For example, `%APPDATA%` *will* be synced over the network/internet on login/logout while `%LOCALAPPDATA%` will not.  By storing files in the correct folders, the application is more likely to "just work" for users working in different environments.

I believe that Linux actually now has [a standard for environment variables](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) similar to the Windows one, so I would encourage fixing Linux at the same time (and cease writing to `$HOME` when those variables are present) but I don't *personally* care about Linux as much.  😄  MacOS X also has a convention I believe, though I'm not sure how this packages behaves in that situation.

**Comment by thaJeztah** [comment]
@simonferquel @lowenna @johnstep @StefanScherer PTAL

**Comment by simonferquel** [comment]
This makes sense but it is a breaking change.
